docs/RELEASES.md

```diff
@@ -0,0 +1,422 @@
+
+- The `beta` branch is merged to `release`. The resulting version on release changes from 8.2 to 9.0.
+- The `main` branch is merged to `beta`. The resulting version on beta changes from 9.0b4 to 10.0b1
+- The `main` branch version number is changed from 10.0a1 to 11.0a1
+
+While the version name changes, it must be ensured that the version code stays the same for each branch. Our application IDs are specific to the branch they are on. For example:
+
+- Beta always uses `net.thunderbird.android.beta` as the app ID. Let's say the version code is 20 at 9.0b4, it will be 21 at 10.0b1.
+- Likewise, when 9.0b4 becomes 9.0, if the version code on beta is 20 and on release it is 12, then 9.0 becomes 13 and not 21.
+
+## Merge Process
+
+Merges are performed with the `git merge` command:
+
+```sh
+ git checkout beta
+ git merge main
+```
+
+This approach enables various benefits, including:
+
+- Carrying forward main branch history to beta, and beta branch history to release.
+- No branch history is lost.
+- Git tags are retained in git log.
+- Files/code that is unique per branch can remain that way (e.g. notes files such as changelog_master.xml, version codes).
+
+## Branch Uplifts
+
+If the urgency of a fix requires it to be included in the Beta or Release channel before the next merge, the uplift process is followed. If possible, uplifts should be avoided and patches should “ride the train” instead, following the merge day cycle.
+
+### Uplift Criteria
+
+Beta uplifts should:
+
+- Be limited to bug/security fixes only (features ride the train).
+- Not change any localizable strings
+- Have tests, or a strong statement of what can be done in the absence of tests.
+- Have landed in main and stabilized on the daily channel.
+- Have a comment in the GitHub issue assessing performance impact, risk, and reasons the patch is needed on beta.
+
+Release uplifts should additionally:
+
+- Be limited to stability/security fixes only (features ride the train).
+- Have landed in beta and stabilized on the beta channel.
+
+### Uplift Process
+
+1. The requestor creates a pull request to the respective target branch with the cherry-picked commits they intend to uplift.
+2. The requestor makes a comment in the bug with the Approval Request Comment template filled out.
+3. The requestor includes a link to the pull request in the approval request comment and sets the `uplift-approval?` label.
+4. The release driver reviews pull requests with the `uplift-approval?` label, and will merge any pull requests that are approved, or close any pull requests that are declined.
+
+Uplift patches are generated with:
+
+```sh
+git cherry-pick -x <commit-hash>
+```
+
+Template for uplift requests:
+
+```sh
+[Approval Request Comment]
+Original Issue/Pull request:
+Regression caused by (issue #):
+User impact if declined:
+Testing completed (on daily, etc.):
+Risk to taking this patch (and alternatives if risky):
+```
+
+## Create K-9 Mail releases
```

Comment:
This section describes the manual release process, mostly. I think it might be useful to keep in some form, let's say the apocalypse happens and GitHub actions are no longer available or certain steps fail, we'd know how to do it manually.

Otherwise I'd rather suggest this section describes how we trigger releases, what manual intervention is necessary, etc.

If you'd rather do this incrementally I'm fine with adding this section as long as we add a NOTE at the top of the section that we do things automatically now.